pkg/runtime/framework/core/framework.go

```diff
@@ -148,6 +155,16 @@ func (f *Framework) RunTerminalConditionPlugins(ctx context.Context, trainJob *t
 	return nil, nil
 }
 
+func (f *Framework) RunJobsStatusPlugins(ctx context.Context, trainJob *trainer.TrainJob) ([]trainer.JobStatus, error) {
+	if len(f.jobsStatusPlugins) > 1 {
+		return nil, errorTooManyJobsStatusPlugin
+	}
+	if len(f.jobsStatusPlugins) != 0 {
+		return f.jobsStatusPlugins[0].JobsStatus(ctx, trainJob)
```

Comment:
@tenzen-y @astefanutti What is the use-case when we can have more than one terminalConditionPlugin or jobsStatusPlugin ?

Reply:
@andreyvelich that would be if two plugins implement the same plugin interface so they would conflict.

The check is currently performed for each call, maybe we could do that during the runtime framework initialization.

Reply:
But I am trying to understand when users can define more than 1 implementation for such plugins ?
Terminal condition or job status should always be defined by one plugin, unless I am missing something ?

Reply:
Conceptually yes, but practically it is possible to have multiple plugins implementing the same interface like `framework.TerminalConditionPlugin`.

We could imagine this happens if we had other backend than JobSet, e.g., LWS and both would be made active by "mistake".

Reply:
> both would be made active by "mistake".

If that is set by mistake, shall we validate that only one plugin can be activated for such interfaces during framework initialization ?

Reply:
> If that is set by mistake, shall we validate that only one plugin can be activated for such interfaces during framework initialization ?

I agree, that's what I meant when I said "maybe we could do that during the runtime framework initialization.".

Reply:
@andreyvelich I've updated the logic for those "singleton" plugins so it fails-fast on at init time.